fpdf/fpdf.py

```diff
@@ -1266,17 +1267,116 @@ def rect(self, x, y, w, h, style=None):
             y (float): Ordinate of upper-left bounding box.
             w (float): Width.
             h (float): Height.
+            round_corners (tuple of str, tuple of fpdf.enums.Corner, bool): Optional draw a rectangle with round corners.
+            Possible values are:
+
+            *`TOP_LEFT`: a rectangle with round top left corner
+            *`TOP_RIGHT`: a rectangle with round top right corner
+            *`BOTTOM_LEFT`: a rectangle with round bottom left corner
+            *`BOTTOM_RIGHT`: a rectangle with round bottom right corner
+            *`True`: a rectangle with all round corners
+            *`False`: a rectangle with no round corners
             style (fpdf.enums.RenderStyle, str): Optional style of rendering. Possible values are:
 
             * `D` or empty string: draw border. This is the default value.
             * `F`: fill
             * `DF` or `FD`: draw and fill
         """
-        style = RenderStyle.coerce(style)
-        self._out(
-            f"{x * self.k:.2f} {(self.h - y) * self.k:.2f} {w * self.k:.2f} "
-            f"{-h * self.k:.2f} re {style.operator}"
-        )
+
+        if round_corners is not False:
+            self._draw_rounded_rect(x, y, w, h, style, round_corners)
+        else:
+            style = RenderStyle.coerce(style)
```

Comment:
This should be called before the `if` to ensure that `style` has a valid value that can be converted to a `RenderStyle`

Reply:
Fixed